src/mount.c

```diff
@@ -36,66 +38,143 @@ pu_create_mount_point(const gchar *name,
     return mount_point;
 }
 
+static gboolean
+pu_find_mounted(const gchar *device,
+                GList **mounted)
+{
+    blkid_probe pr;
+    blkid_partlist ls;
+    gint i, num_partitions;
+    struct libmnt_table *tb;
+    g_autofree gchar *partname;
+    g_autoptr(GList) list = NULL;
+    g_autoptr(GError) error = NULL;
+
+    g_return_val_if_fail(g_strcmp0(device, "") > 0, FALSE);
+
+    pr = blkid_new_probe_from_filename(device);
+    if (!pr)
+        return FALSE;
+
+    ls = blkid_probe_get_partitions(pr);
+    if (!ls) {
+        blkid_free_probe(pr);
+        return FALSE;
+    }
+    num_partitions = blkid_partlist_numof_partitions(ls);
+    if (num_partitions < 0) {
+        blkid_free_probe(pr);
+        return FALSE;
+    }
+    blkid_free_probe(pr);
+
+    tb = mnt_new_table();
+    if (!tb)
+        return FALSE;
+
+    if (mnt_table_parse_mtab(tb, NULL) < 0) {
+        mnt_unref_table(tb);
+        return FALSE;
+    }
+
+    for (i = 0; i < num_partitions; i++) {
+        partname = pu_device_get_partition_path(device, i + 1, &error);
+        if (!partname) {
+            g_printerr("Failed with error message: %s", error->message);
+            mnt_unref_table(tb);
+            return FALSE;
+        }
+        if (mnt_table_find_source(tb, partname, MNT_ITER_FORWARD))
+            list = g_list_prepend(list, partname);
+    }
+    *mounted = list;
```

Comment:
Because `list` is being freed automatically after returning, I think you have to use `g_steal_pointer()` here to transfer ownership to `mounted`. Please have a look at the documentation yourself though, just to be sure on this matter.

Reply:
I looked it up and you're right `g_steal_pointer()` must be used here.